### PR TITLE
chore: adds CA2007 warning

### LINKED ISSUE #168: Consider enabling CA2007 rule to require `ConfigureAwait(false)` on all awaited tasks
### What is the version or commit of the ORAS .NET library?

_No response_

### What would you like to be added?

I would like the .editorconfig to be modified to include the CA2007 rule, which requires awaited tasks to have `ConfigureAwait(false)`

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2007

### Why is this needed for the ORAS .NET library?

This would make reviewing PRs easier since we, as a library, must `ConfigureAwait(false)` everywhere.

Sample PR implementing such changes:
https://github.com/dotnet/aspnetcore/pull/39946

Confirm we want it and I'll create the PR :+1: 

### Are you willing to submit PRs to contribute to this feature?

- [X] Yes, I am willing to implement it.

### PR DESCRIPTION
**Pull request by leonardochaia**
### What this PR does / why we need it

This PR adds severity Warning when `ConfigureAwait(false)` is not used in an awaited `Task`.
This is excluded on Tests project.

### Which issue(s) this PR resolves / fixes

Fixes #168 

### Please check the following list
- [ ] Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ] Does this change require a documentation update?
- [ ] Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ] Do all new files have an appropriate license header?

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -50,4 +50,7 @@ dotnet_naming_symbols.public_members.applicable_kinds = property, field, method
 dotnet_naming_symbols.public_members.applicable_accessibilities = public
 dotnet_naming_symbols.public_members.required_modifiers = 
 
-dotnet_naming_style.public_uppercase_style.capitalization = pascal_case
\ No newline at end of file
+dotnet_naming_style.public_uppercase_style.capitalization = pascal_case
+
+# CA2007: Consider calling ConfigureAwait on the awaited task
+dotnet_diagnostic.CA2007.severity = warning
```
Comment (Wwwsylvia) [inline .editorconfig:56]: nit: Shall we include an empty line at the end of file?

**Review by Wwwsylvia — APPROVED** [review]
LGTM with nit

File: .editorconfig (hunk 1 above)
Comment (leonardochaia) [reply .editorconfig:56]: Yep. Will do :+1:

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**